### PR TITLE
Move to sdk 21

### PR DESCRIPTION
**Pull request by kordianbruck**
## Issue

This fixes the following issue(s): #711 

@jacqueline8711 If you see any more simplifications that can be done with the eduroam code, go ahead.

### CODE REVIEW

**Comment by pfent** [comment]
@kordianbruck can we do a Bugfix Release for that wifi crash prior to merging this?

**Comment by kordianbruck** [comment]
@pfent I'm merging this to the 1.6 release branch - no worries. Current master will become 1.5 release branch

**Comment by pfent** [comment]
ah, now we're doing release branches again... I'm certainly not a fan, since this usually creates more problems than it solves.

> Current master will become 1.5 release branch

then why don't branch release/1.5 and continue developing on master? That's how the kernel does it and what usually works best with git...

**Comment by kordianbruck** [comment]
@pfent still waiting for some bug fixes. Will certainly switch it around, once I do the release.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/eduroam/EduroamFixCard.java [hunk 1]
```diff
@@ -123,69 +125,68 @@ private boolean isConfigValid() {
...
+            checkDNSName();
+            checkAnonymousIdentity();
+            // note: checking the certificate does not seem possible
         }
+        // else: PWD or unknown authentication method (we don't know if that method is safe or not -> ignore)
+
         return errors.isEmpty();
     }
 
     @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/eduroam/EduroamFixCard.java]: This RequiresApi can be removed, when we move to 21

**Comment by pfent** [comment]
And since we're moving to API 21, consider removing dead code in:
- [ ] FavoriteDishAlarmScheduler
- [ ] ScanResultsAvailableReceiver
- [ ] Utils
- [ ] StartSyncReceiver
- [ ] IntegratedCalendarEvent
- [ ] EduroamController 
- [ ] SetupEduroamActivity
- [ ] TimetableWidgetService

just search for `Build.VERSION`

There are also some `@SuppressLint("NewApi")` and `@SuppressLint("InlinedApi")` Annotations, that need review

**Comment by jacqueline8711** [comment]
@kordianbruck sorry this is such a mess
I was just trying to fix my username for the first three commits... 😬

**Comment by kordianbruck** [comment]
@jacqueline8711 no worries :)

**Comment by pfent** [comment]
Also, when moving to 21, we could also drop the PDFView dependency:

>Also, an inline display for the PDFs would be nice to have. There is a [PDFRenderer](https://developer.android.com/reference/android/graphics/pdf/PdfRenderer.html) available for API 21+, even with an exhaustive [example on GitHub](https://github.com/googlesamples/android-PdfRendererBasic). Then you can use the fileprovider as a fallback for lower APIs.

[AndroidPdfViewer](https://github.com/barteksc/AndroidPdfViewer) comes with PdfiumAndroid, which is ~16MB

**Pull request merged**